html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -163,22 +170,150 @@ public function home()
+				$bidinfo_array = $this->Bidinfo->find('all')->toArray();
+				$count = count($this->Bidinfo->find('all')->toArray());
+				for ($i = 0; $i <= $count; $i++) {
+					if ($bidinfo_array[$i]['biditem_id'] === $entity["biditem_id"]) {
+						$bidinfo_id = $bidinfo_array[$i]['id']; //bidinfoのid
+						break;
+					}
+				}
+				$bidmsg = $this->Bidmessages->newEntity();
+				$bidmsgs = ['bidinfo_id' => $bidinfo_id, 'user_id' => $entity["buyer_id"], 'message' => '落札者が商品を受け取りました'];
+				$bidmsg = $this->Bidmessages->patchEntity($bidmsg, $bidmsgs);
+				// Bidmessageに発送完了を保存
+				$this->Bidmessages->save($bidmsg);
+				return $this->redirect(['action' => 'msg', $bidinfo_array[$i]['id']]);
+			}
+		}
+		$buyer_status = $this->BuyerStatus->find('all')->toArray();
+		$ratings = $this->Ratings->find('all')->toArray();
+		$this->set(compact('buyer_status', 'ratings'));
 	}
 
 	// 出品情報の表示
 	public function home2()
 	{
 		// 自分が出品したBiditemをページネーションで取得
 		$biditems = $this->paginate('Biditems', [
-			'conditions'=>['Biditems.user_id'=>$this->Auth->user('id')], 
+			'conditions' => ['Biditems.user_id' => $this->Auth->user('id')],
 			'contain' => ['Users', 'Bidinfo'],
-			'order'=>['created'=>'desc'],
-			'limit' => 10])->toArray();
+			'order' => ['created' => 'desc'],
+			'limit' => 10
+		])->toArray();
 		$this->set(compact('biditems'));
+
+		if ($this->request->is('post')) {
+			$biditems_id = (intval($this->request->getData(['item'])));
+			$entity = $this->Biditems->get($biditems_id);
+			$this->Biditems->patchEntity($entity, ['is_sent' => true]);
+			$this->Biditems->save($entity);
+			if ($this->Biditems->save($entity)) {
+				$bidinfo_array = $this->Bidinfo->find('all')->toArray();
+				$count = count($bidinfo_array);
+				for ($i = 0; $i <= $count; $i++) {
+					if ($bidinfo_array[$i]['biditem_id'] === $entity["id"]) {
+						$bidinfo_id = $bidinfo_array[$i]['id']; //bidinfoのid
+						break;
+					}
+				}
+				$bidmsg = $this->Bidmessages->newEntity();
+				$bidmsgs = ['bidinfo_id' => $bidinfo_id, 'user_id' => $entity["user_id"], 'message' => '出品者から商品が発送されました。'];
+				$bidmsg = $this->Bidmessages->patchEntity($bidmsg, $bidmsgs);
+				// Bidmessageに発送完了を保存
+				$this->Bidmessages->save($bidmsg);
+				return $this->redirect(['action' => 'msg', $bidinfo_array[$i]['id']]);
+			}
+		}
+		$ratings = $this->Ratings->find('all')->toArray();
+		$buyer_status = $this->BuyerStatus->find('all')->toArray();
+		$this->set(compact('ratings', 'buyer_status'));
+	}
+	// 落札者情報
+	public function buyerinfo()
+	{
+		// 落札者とログインユーザーが一緒か、商品idが落札の商品idと一緒か==================================
+		// 自分が落札したBidinfoをページネーションで取得
+		$bidinfos = $this->paginate('Bidinfo', [
+			'conditions' => ['Bidinfo.user_id' => $this->Auth->user('id')],
+			'contain' => ['Users', 'Biditems']
+		])->toArray();
+
+		$query_parameter_isError = false;
+		foreach ($bidinfos as $bidinfo) {
+			if ($bidinfo->biditem_id === intval($_GET["biditem_id"]) && $bidinfo->user_id === $this->Auth->user('id')) {
```

Comment:
conditionsでuser_idで絞っているためこの比較は常に真になるので不要ですね
```suggestion
			if ($bidinfo->biditem_id === intval($_GET["biditem_id"])) {
```